packages/core/src/common/abstractPureComponent2.ts

```diff
@@ -16,12 +16,15 @@
 
 import * as React from "react";
 import { isNodeEnv } from "./utils";
+import { IWindowOverrideContext, WINDOW_OVERRIDE_REACT_CONTEXT_TYPES } from "./windowOverrideContext";
 
 /**
  * An abstract component that Blueprint components can extend
  * in order to add some common functionality like runtime props validation.
  */
 export abstract class AbstractPureComponent2<P, S = {}, SS = {}> extends React.PureComponent<P, S, SS> {
+    public static contextTypes = WINDOW_OVERRIDE_REACT_CONTEXT_TYPES;
+
     // unsafe lifecycle method
     public componentWillUpdate: never;
     public componentWillReceiveProps: never;
```

Comment:
I think I had a reason for not declaring context's type. I'm not sure of the best way to handle the way inheritance works here. Do we want children to have the same context type? Should we even do the contextTypes thing?